### PR TITLE
Fix for mac-os permissions pre-commit

### PR DESCRIPTION
**Pull request by Jolanrensen**
Fixes https://github.com/Kotlin/dataframe/issues/612

### CODE REVIEW

File: core/build.gradle.kts [hunk 1]
```diff
@@ -173,6 +174,12 @@ val installGitPreCommitHook by tasks.creating(Copy::class) {
         from(File(rootProject.rootDir, "gradle/scripts/pre-commit"))
         into(gitHooksDir)
         fileMode = 755
+
+        // Workaround for https://github.com/Kotlin/dataframe/issues/612
+        if (OSType.identify() == OSType.Mac) exec {
```
[the remark this answers is not in the document]
Comment (Jolanrensen) [reply core/build.gradle.kts:179]: good point

**Comment by Jolanrensen** [comment]
@devcrocod Could you check the latest version of this branch on your mac?

**Comment by koperagen** [comment]
I can confirm that chmod does it job, but i don't like how that hook works now :c I tried to commit just one file, but it added all other uncommited files and delay was quite evident too

**Comment by Jolanrensen** [comment]
> I can confirm that chmod does its job, but i don't like how that hook works now :c I tried to commit just one file, but it added all other uncommited files and delay was quite evident too

I agree. It's mostly there so newly generated files are added to the git index automatically, because people forget. But it should mirror the commits to `src` only.

**Review by devcrocod — APPROVED** [review]
This works for me. I don't particularly like this solution for aesthetic reasons, but it works.

**Pull request merged**